tests/buffer/buffer_data_tracking.cpp

```diff
@@ -11,44 +13,148 @@ using namespace cl::sycl;
 constexpr size_t N = 3;
 using Vector = float[N];
 
-int main() {
-  Vector a = { 1, 2, 3 };
-
-  boost::compute::context context { boost::compute::system::default_context() };
-  boost::compute::command_queue b_queue { context, context.get_device() };
-  auto program = boost::compute::program::create_with_source(R"(
-    __kernel void zeroify(__global int *a) {
-      a[get_global_id(0)] = 0;
-    }
-    )", context);
+Vector a = { 1, 2, 3 };
 
-  program.build();
+void test1(auto program, auto b_queue) {
```

Comment:
I had a dream... Maybe in C++20?
in the meantime you have to write:
```
auto test1 = [](auto program, auto b_queue) {
```
Have you run this test by the way?
`tests/buffer/read_write_buffer.cpp` fails too... :-(
```